task/servicetest/servicetest.go

```diff
@@ -1437,3 +1443,87 @@ option task = {
 from(bucket: "b")
 	|> http.to(url: "http://example.com")`
 )
+
+func testTaskType(t *testing.T, sys *System) {
+	cr := creds(t, sys)
+	authorizedCtx := icontext.SetAuthorizer(sys.Ctx, cr.Authorizer())
+
+	// Create a tasks
+	ts := influxdb.TaskCreate{
+		OrganizationID: cr.OrgID,
+		Flux:           fmt.Sprintf(scriptFmt, 0),
+		Token:          cr.Token,
+	}
+
+	tsk, err := sys.TaskService.CreateTask(authorizedCtx, ts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !tsk.ID.Valid() {
+		t.Fatal("no task ID set")
+	}
+
+	tc := influxdb.TaskCreate{
+		Type:           "cows",
+		OrganizationID: cr.OrgID,
+		Flux:           fmt.Sprintf(scriptFmt, 0),
+		Token:          cr.Token,
+	}
+
+	tskCow, err := sys.TaskService.CreateTask(authorizedCtx, tc)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !tskCow.ID.Valid() {
+		t.Fatal("no task ID set")
+	}
+
+	tp := influxdb.TaskCreate{
+		Type:           "pigs",
+		OrganizationID: cr.OrgID,
+		Flux:           fmt.Sprintf(scriptFmt, 0),
+		Token:          cr.Token,
+	}
+
+	tskPig, err := sys.TaskService.CreateTask(authorizedCtx, tp)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !tskPig.ID.Valid() {
+		t.Fatal("no task ID set")
+	}
+
+	// get default tasks
+	tasks, _, err := sys.TaskService.FindTasks(sys.Ctx, influxdb.TaskFilter{OrganizationID: &cr.OrgID})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, task := range tasks {
+		if task.Type != "" {
+			t.Fatal("recieved a task with a type when sending no type restriction")
+		}
+	}
+
+	// get filtered tasks
+	tasks, _, err = sys.TaskService.FindTasks(sys.Ctx, influxdb.TaskFilter{OrganizationID: &cr.OrgID, Type: &tc.Type})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(tasks) != 1 {
+		t.Fatalf("failed to return tasks by type, expected 1, got %d", len(tasks))
+	}
+
+	// get all tasks
+	wc := influxdb.TaskTypeWildcard
+	tasks, _, err = sys.TaskService.FindTasks(sys.Ctx, influxdb.TaskFilter{OrganizationID: &cr.OrgID, Type: &wc})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// this is to account for the concurrency test creating tasks in the same organization
```

Comment:
Another take it or leave it:

If you want to get precise here you could update the CredsFunc to take a *testing.T here: https://github.com/influxdata/influxdb/blob/c4fd4d79865873795b02dbdd1bcf26b3fd3e6fcd/http/task_test.go#L70

And then move org creation into the creds func call as well.
Then you can properly scope the organisation to the subtest name. That way you don't share organisations between parallel test runs leading to unpredictable results. Could avoid someone in the future deleting that concurrency test and breaking this one for example.
Also this looks like only happens in the http/task_test.go as that defines its own CredFunc override which causes this problem.

Reply:
woo fresh eyes are so nice. Thanks! that little change will be very good for this testing suite.